WMF Framework/PermanentCacheController.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+
+@objc(WMFPermanentCacheController)
+public final class PermanentCacheController: NSObject {
+    public let imageCache: ImageCacheController
+    public let articleCache: ArticleCacheController
+    let urlCache: PermanentlyPersistableURLCache
+    let managedObjectContext: NSManagedObjectContext
+    
+    /// - Parameter moc: the managed object context for the cache
+    /// - Parameter session: the session to utilize for image and article requests
+    /// - Parameter configuration: the configuration to utilize for configuring requests
+    /// - Parameter preferredLanguageDelegate: the preferredLanguageDelegate to utilize for determining the user's preferred languages
+    @objc public init(moc: NSManagedObjectContext, session: Session, configuration: Configuration, preferredLanguageDelegate: WMFPreferredLanguageCodesProviding) {
+        imageCache = ImageCacheController(moc: moc, session: session, configuration: configuration)
+        articleCache = ArticleCacheController(moc: moc, imageCacheController: imageCache, session: session, configuration: configuration, preferredLanguageDelegate: preferredLanguageDelegate)
+        urlCache = PermanentlyPersistableURLCache(moc: moc)
+        managedObjectContext = moc
+        super.init()
+        session.permanentCache = self
+    }
+    
+    /// Performs any necessary migrations on the CacheController's internal storage
+    /// Exists on this @objc ImageCacheControllerWrapper so it can be accessed from WMFDataStore
+    @objc public static func setupCoreDataStack(_ completion: @escaping (NSManagedObjectContext?, Error?) -> Void) {
+        CacheController.setupCoreDataStack(completion)
+    }
+    
+    @objc public func fetchImage(withURL url: URL?, failure: @escaping (Error) -> Void, success: @escaping (ImageDownload) -> Void) {
+        let _ = imageCache.fetchImage(withURL: url, failure: failure, success: success)
+    }
+    
+    @objc public func memoryCachedImage(withURL url: URL) -> Image? {
+        return imageCache.memoryCachedImage(withURL: url)
+    }
+    
+    @objc public func fetchImage(withURL url: URL?, priority: Float, failure: @escaping (Error) -> Void, success: @escaping (ImageDownload) -> Void) -> String? {
+        return imageCache.fetchImage(withURL: url, priority: priority, failure: failure, success: success)
+    }
+    
+    @objc public func cancelFetch(withURL url: URL?, token: String?) {
```

Comment:
Since this passthrough method only operates on the `imageCache` within, I think we should name it `cancelImageFetch` just to be more clear to outside callers.